src/Microsoft.Extensions.Caching.Memory/IMemoryCacheEvictionStrategy.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Caching.Memory
+{
+    public interface IMemoryCacheEvictionStrategy
+    {
+        // TODO: doc comments
+        IEnumerable<CacheEntry> GetEntriesToEvict(IEnumerable<CacheEntry> entries, DateTimeOffset now);
```

Comment:
Should this be `IList<CacheEntry>`? This structure worries me a little because it requires the caller to do a LINQ query (or some equivalent) by default. We should add a micro benchmark to see what the performance overhead of this is.